src/server/rdb_load.cc

```diff
@@ -1874,7 +1878,27 @@ auto RdbLoaderBase::ReadSBF() -> io::Result<OpaqueObj> {
     unsigned hash_cnt;
     string filter_data;
     SET_OR_UNEXPECT(LoadLen(nullptr), hash_cnt);
-    SET_OR_UNEXPECT(FetchGenericString(), filter_data);
+
+    if (absl::GetFlag(FLAGS_rdb_sbf_chunked)) {
```

Comment:
I do not think you need to use `FLAGS_rdb_sbf_chunked` in the loader.
The load part should only rely on `RDB_TYPE_SBF2` and `RDB_TYPE_SBF`  to decide how to parse. it should be stateless in this sense.